slate-note-taking/src/Components/Context/LoginSignupContext.js

```diff
@@ -0,0 +1,100 @@
+import axios from "axios";
+import React, { createContext, useContext, useReducer } from "react";
+import { useNavigate } from "react-router-dom";
+import Toast from "../Toast/Toast";
+
+const loginSignupContext = createContext();
+export const useLoginSignupContext = () => useContext(loginSignupContext);
+
+function LoginSignupContext({ children }) {
+  const navigate = useNavigate();
+
+  const [state, dispatch] = useReducer(reducerFn, {
+    name: "",
+    email: "",
+    number: "",
+    password: "",
+    loginData: [],
+  });
+  function reducerFn(state, action) {
+    switch (action.type) {
+      case "NAME":
+        return { ...state, name: action.payload };
+      case "EMAIL":
+        return { ...state, email: action.payload };
+      case "NUMBER":
+        return { ...state, number: action.payload };
+      case "PASSWORD":
+        return { ...state, password: action.payload };
+      case "LOGINDATA":
+        return { ...state, loginData: action.payload };
+      default:
+        return state;
+    }
+  }
+
+  const { name, email, password, number, loginData } = state;
+  console.log(
+    "🚀 ~ file: LoginSignupContext.js ~ line 37 ~ LoginSignupContext ~ loginData",
+    loginData
+  );
+  const loginHandler = async (e) => {
+    e.preventDefault();
+    try {
+      const response = await axios.post(`/api/auth/login`, {
+        email: email,
+        password: password,
+      });
+      // saving the encodedToken in the localStorage
+      dispatch({ type: "LOGINDATA", payload: response.data.foundUser });
+      localStorage.getItem(`token`, response.data.encodedToken);
```

Comment:
sorry if im wrong but i think its supposed to be a setter not a getter.
suggestion: use optional chaining here